heft-plugins/heft-sass-plugin/src/SassProcessor.ts

```diff
@@ -194,39 +196,46 @@ export class SassProcessor extends StringValuesTypingsGenerator {
     buildFolder: string,
     importIncludePaths: string[] | undefined
   ): Promise<string> {
-    let result: Result;
+    let result: CompileResult;
+    const nodeModulesUrl: URL = pathToFileURL(`${buildFolder}/node_modules/`);
     try {
-      result = await LegacyAdapters.convertCallbackToPromise(render, {
-        data: fileContents,
-        file: filePath,
-        importer: (url: string) => ({ file: this._patchSassUrl(url) }),
-        includePaths: importIncludePaths
+      result = compileString(fileContents, {
```

Comment:
Does this have to be synchronous? I assume this is calling some native code and waiting for a result, so it'd be nice to not have to block the JS thread.

Reply:
Nevermind. Looks like that API is slower.

Reply:
Sass API documentation says that the synchronous API is almost twice as fast as the asynchronous one.
(ref: https://sass-lang.com/documentation/js-api/modules#compileStringAsync)

To be honest, I haven't compared this directly, but the compilation job is a CPU intensive operation in general, I think the benefit of completing the compilation faster is greater than the benefit of calling a kernel function asynchronously I think.

Nevertheless, if you still consider this to work asynchronously, I'd be happy to fix it.